src/jtag/drivers/ftdi_cjtag.c

```diff
@@ -0,0 +1,787 @@
+*   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+*   GNU General Public License for more details.                          *
+*                                                                         *
+*   You should have received a copy of the GNU General Public License     *
+*   along with this program; if not, write to the                         *
+*   Free Software Foundation, Inc.,                                       *
+*   51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.           *
+***************************************************************************/
+
+/**
+* @file
+* Handles cJTAG comms to the ARC processor.
+*/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+/* project specific includes */
+#include <jtag/interface.h>
+#include <jtag/swd.h>
+#include <transport/transport.h>
+#include <helper/time_support.h>
+
+#if IS_CYGWIN == 1
+#include <windows.h>
+#endif
+
+#include <assert.h>
+
+/* FTDI access library includes */
+#include "mpsse.h"
+
+#define TDI_0 0 
+#define TDI_1 1
+#define TMS_0 0 
+#define TMS_1 2 
+
+#define STFMT 3 /*  jtag7 store format command. */
+#define SCNS 9  /* jtag7 scan string */
+#define STMC 0  /* jtag7 Store Miscellaneous Control */
+#define MSC 7   /* jtag7 make scan group candidate */
+
+static const unsigned OSCAN1 = 9;   /* jtag7 format=oscan1 parameter. */
+static const unsigned EXIT_CMD_LVL = 1;
+
+typedef enum
+{
+	normal, 
+	oscan1
+} TycJtagScanFormat;
+
+#define MAX_SUPPORTED_SCAN_LENGTH 1000
+static unsigned char pucRxBuffer[MAX_SUPPORTED_SCAN_LENGTH];
+#define MAX_PREAMBLE  6
+#define MAX_POSTAMBLE 2
+
+TycJtagScanFormat tycJtagScanFormat = normal;
+static int iWantResult  = 1;
+static unsigned long ulTxCount = 0;
+static void enter_2wire_mode(void);
+static int want_result(int wr);
+static void set_format(TycJtagScanFormat sf);
+static bool in_normal_scan_mode(void);
+static int enter_2wire_mode_and_test(void);
+static void enter_oscan1(void);
+static void check_packet(void);
+static void send_check_packet(void);
+static void dummy_scan_packet(void);
+static void send_JTAG(unsigned tms_and_tdi);
+static void send_2part_command(unsigned opcode, unsigned operand);
+static void enter_control_mode(int level) ;
+static void non_zero_bit_scan_sequence(unsigned amount);
+static void zero_bit_scan_sequence(int repeat);
+static void transitions(const char *tms_values, int tdi_value);
+static void set_JTAG_to_idle(void);
+static void cjtag_move_to_state(tap_state_t goal_state);
+
+static struct mpsse_ctx *local_ctx;
+
+static const unsigned char pucInitializeSequence[] = {	0x80, /* Set output pins			*/
+														0xE8, /*		Value				*/
+														0xEB, /*		Direction			*/
+														0x82, /* Set output pins (high byte)*/
+														0x00, /*		Value				*/
+														0x60, /*		Direction			*/
+														0x81, /* Read input pins			*/
+														0x87};/* Send immediate				*/
```

Comment:
That doesn't look good, when tabs are not 4-space wide :) I believe, that correct formatting would be 
```c
static const unsigned char pucInitializeSequence[] = {
	0x80, /* Set output pins			*/
	0xE8, /*		Value				*/
....
};
```

For example: https://github.com/foss-for-synopsys-dwc-arc-processors/openocd/blob/arc-0.9-dev-2017.03/src/target/mips32.c#L622